src/controllers/expenses.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const expenseService = require('../services/expenses');
+const userService = require('../services/users');
+
+const getAll = (req, res) => {
+  const {
+    userId,
+    categories,
+    from,
+    to,
+  } = req.query;
+
+  if (userId) {
+    if (!Number.isInteger(+userId)) {
+      res.sendStatus(422);
+
+      return;
+    }
+  }
+
+  if (categories) {
+    const isCategoriesValid = typeof categories === 'string'
+      || Array.isArray(categories);
+
+    if (!isCategoriesValid) {
+      res.sendStatus(422);
+
+      return;
+    }
+  }
+
+  const expenses = expenseService.getAll(userId, categories, from, to);
+
+  res.send(expenses);
+};
+
+const getOne = (req, res) => {
+  const { expenseId } = req.params;
+  const foundExpense = expenseService.getById(expenseId);
+
+  if (!foundExpense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(foundExpense);
+};
+
+const add = (req, res) => {
```

Comment:
I think, it would be better name
```suggestion
const create = (req, res) => {
```